source/cellular_bg96_api.c

```diff
@@ -1700,21 +1780,41 @@ static CellularPktStatus_t _Cellular_RecvFuncGetPsmSettings( CellularContext_t *
 }
 
 /*-----------------------------------------------------------*/
+static CellularPktStatus_t socketRecvDataPrefixSsl( void * pCallbackContext,
+                                                 char * pLine,
+                                                 uint32_t lineLength,
+                                                 char ** ppDataStart,
+                                                 uint32_t * pDataLength )
+{
+    return socketRecvDataPrefix( pCallbackContext, pLine, lineLength, ppDataStart, pDataLength, true );
+}
 
-static CellularPktStatus_t socketRecvDataPrefix( void * pCallbackContext,
+static CellularPktStatus_t socketRecvDataPrefixNonSsl( void * pCallbackContext,
                                                  char * pLine,
                                                  uint32_t lineLength,
                                                  char ** ppDataStart,
                                                  uint32_t * pDataLength )
+{
+    return socketRecvDataPrefix( pCallbackContext, pLine, lineLength, ppDataStart, pDataLength, false);
+}
+static CellularPktStatus_t socketRecvDataPrefix( void * pCallbackContext,
+                                                 char * pLine,
+                                                 uint32_t lineLength,
+                                                 char ** ppDataStart,
+                                                 uint32_t * pDataLength,
+                                                 bool useSsl )
 {
     char * pDataStart = NULL;
     uint32_t prefixLineLength = 0U;
     int32_t tempValue = 0;
     CellularATError_t atResult = CELLULAR_AT_SUCCESS;
     CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
     uint32_t i = 0;
-    char pLocalLine[ MAX_QIRD_STRING_PREFIX_STRING + 1 ] = "\0";
-    uint32_t localLineLength = MAX_QIRD_STRING_PREFIX_STRING > lineLength ? lineLength : MAX_QIRD_STRING_PREFIX_STRING;
+    uint8_t commandPrefixStringLength = useSsl ? MAX_QSSLRECV_STRING_PREFIX_STRING : MAX_QIRD_STRING_PREFIX_STRING;
+    uint8_t sslDataPrefixStringLength = useSsl ? SSL_DATA_PREFIX_STRING_LENGTH : DATA_PREFIX_STRING_LENGTH;
+    char pLocalLine[ commandPrefixStringLength + 1 ];

```

Comment:
I don't think this is safe. The unititialized char array might not have a \0, and strcpy will fail.

```suggestion
    char pLocalLine[ commandPrefixStringLength + 1 ] = "\0";
```

Reply:
I don't understand the logic in your suggestion, sorry. I just maintain the code provided by the FreeRTOS team and changed the commandPrefixStringLength to include the SSL implementation